docs/content/sips/017-service-chaining.md

```diff
@@ -0,0 +1,106 @@
+
+**DECISION REQUIRED:** Should `allowed_outbound_hosts` including _either_ of `self` or `spin.internal` be enough to grant permission for self-requests using chaining?
+
+Currently, `self` allows access to all routes in the application.  The initial version of service chaining will be similar; if someone needs to restrict access to specific components, that can be a future option.
+
+# Runtime and framework considerations
+
+## Chaining requirements in the lock file
+
+The lock file format built into existing hosts is designed for extension, and so [avoids using serde "deny unknown fields"](https://github.com/fermyon/spin/blob/663593d1423ed3518744d6f797e6a3970575d617/crates/locked-app/src/locked.rs#L16).  Unfortunately, this means there is no way to add a prerequisites section to the lock file in such a way that existing hosts will reject it if present.  For example, if someone deploys a chaining-required app onto an older version of the containerd shim, it will appear to accept it, but will not chain.
+
+Unfortunately, I don't think there's anything we can do about this, except to introduce a new `spin_lock_version`.  Doing this naively would mean existing hosts would reject the lock file _whether or not it contains prerequisites_.  We could get smart and emit a v.next lock file _only_ if the app expressed prerequisites but hoo boy.
+
+**DECISION REQUIRED:** Should we rev `spin_lock_version`?
+
+In any case, whether as a section or as an entry in `metadata`, the updated format specifies (and the Spin crate enforces) must-understand semantics for all entries under it.
+
+## Spin command line implementation
+
+## Where do invoked components run?
+
+In Spin, all components of an app run within a trigger process (typically a child Spin process, but can also be a plugin).
+
+When the calling component is on the HTTP trigger, life is (relaTively) easy, because the host process has all the HTTP components - that is, all possible invocation targets - loaded and ready to go.
+
+When the calling command is on a _non-HTTP_ trigger, things are more challenging.  Either we must host a copy of the HTTP trigger engine (and load HTTP components) in the non-HTTP trigger process, or we must transfer the request to the HTTP trigger for handling.  Neither of these options is very appealing.
+
+**DECISION REQUIRED:** Do we allow service chaining from non-HTTP components to HTTP ones, and if so where do we run the HTTP components?
```

Comment:
I think the entire premise of this feature probably requires that the chunk of the http trigger code responsible for invoking guest http handlers be extracted into another crate (probably `spin-http` if that doesn't cause some dependency tree meltdown). That chunk can then be reused between the http trigger and the outbound http host component.